pkg/frontend/admin_openshiftcluster_vmresize.go

```diff
@@ -0,0 +1,183 @@
+	vars := mux.Vars(r)
+
+	vmName := r.URL.Query().Get("vmName")
+	err := validateAdminVMName(vmName)
+	if err != nil {
+		return err
+	}
+
+	resourceID := strings.TrimPrefix(r.URL.Path, "/admin")
+
+	doc, err := f.dbOpenShiftClusters.Get(ctx, resourceID)
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		return api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "",
+			"The Resource '%s/%s' under resource group '%s' was not found.",
+			vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
+	case err != nil:
+		return err
+	}
+
+	subscriptionDoc, err := f.getSubscriptionDocument(ctx, doc.Key)
+	if err != nil {
+		return err
+	}
+
+	vmSize := r.URL.Query().Get("vmSize")
+	err = validateAdminVMSize(vmSize)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("called resize vm with params vmName: '%s', vmSize: '%s'", vmName, vmSize)
+
+	a, err := f.azureActionsFactory(log, f.env, doc.OpenShiftCluster, subscriptionDoc)
+	if err != nil {
+		return err
+	}
+
+	_, _ = a.VMSizeList(ctx)
+
+	k, err := f.kubeActionsFactory(log, f.env, doc.OpenShiftCluster)
+	if err != nil {
+		return err
+	}
+
+	nodeList, err := k.KubeList(ctx, "node", "")
+	if err != nil {
+		return err
+	}
+
+	var u unstructured.Unstructured
+	var nodes corev1.NodeList
+	if err = json.Unmarshal(nodeList, &u); err != nil {
+		return err
+	}
+
+	err = kruntime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &nodes)
+	if err != nil {
+		return err
+	}
+
+	var resizeNode *corev1.Node
+	for _, node := range nodes.Items {
+		if _, ok := node.ObjectMeta.Labels["node-role.kubernetes.io/master"]; !ok {
+			continue
+		}
+
+		if strings.EqualFold(vmName, node.ObjectMeta.Name) {
+			resizeNode = node.DeepCopy()
+		}
+
+	}
+
+	if resizeNode == nil {
+		return api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeNotFound, "",
+			"The master node '%s' under resource group '%s' was not found.",
+			vmName, vars["resourceGroupName"])
+	}
+
+	log.Infof("cordoning node '%s'", resizeNode.ObjectMeta.Name)
+	resizeNode.Spec.Unschedulable = true
+	resizeNode.Status = corev1.NodeStatus{}
+	unstruct := &unstructured.Unstructured{}
+	uMap, err := kruntime.DefaultUnstructuredConverter.ToUnstructured(resizeNode)
+	if err != nil {
+		return err
+	}
+	unstruct.Object = uMap
+
+	err = k.KubeCreateOrUpdate(ctx, unstruct)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("draining node '%s'", vmName)
+	err = k.KubeDrain(ctx, vmName)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("stopping node '%s'", resizeNode.ObjectMeta.Name)
+	err = a.VMStopAndWait(ctx, vmName)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("resizing node '%s'", resizeNode.ObjectMeta.Name)
+	err = a.VMResize(ctx, vmName, vmSize)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("starting node '%s'", resizeNode.ObjectMeta.Name)
+	err = a.VMStartAndWait(ctx, vmName)
+	if err != nil {
+		return err
+	}
```

Comment:
This looks like a very long operation. I know that we have VM redeploy already. I'm a bit concerned that we add more and more actions like this into front-end. But I do not have a good idea how to deal with it.

Reply:
I agree it is a long operation, but this was the best idea in terms of making sure all of these actions are handled sequentially in order to allow for proper reconciliation of the preceding action.

Reply:
@m1kola we could break it up into singular geneva actions with reliance of the previous such as.  Thoughts?
1. Drain node
2. Power off node
3. Resize node 
4. Power on node

Reply:
-1 to Long Running Operations (LOR)
Lets break it up.

Reply:
@bennerv @jhoreman I think CloudFit are taking the approach of introducing small generic "building block" endpoints and combining them into bigger things. I think it makes sense.

Reply:
New proposed workflow (actions broken up to support this flow)

(Optional) Cordon and drain through update to K8s object
Stop VM
Resize VM
Start VM
(Optional) Mark as schedulable through update to K8s object. 

TODO: Finish unit tests for resize action